tensorboard/plugins/pr_curve/pr_curves_plugin_test.py

```diff
@@ -0,0 +1,187 @@
+        ['red/pr_curves', 'green/pr_curves', 'blue/pr_curves'],
+        tags_response['colors'])
+    self.assertItemsEqual(
+        ['red/pr_curves', 'green/pr_curves', 'blue/pr_curves'],
+        tags_response['mask_every_other_prediction'])
+
+  def testAvailableSteps(self):
+    """Tests that runs are mapped to correct available steps."""
+    self.assertDictEqual({
+        'colors': [0, 1, 2],
+        'mask_every_other_prediction': [0, 1, 2],
+    }, self.plugin.available_steps_impl())
+
+  def testPrCurvesDataCorrect(self):
+    """Tests that responses for PR curves for run-tag combos are correct."""
+    pr_curves_response = self.plugin.pr_curves_impl(
+        ['colors', 'mask_every_other_prediction'], 'blue/pr_curves')
+
+    # Assert that the runs are correct.
+    self.assertItemsEqual(
+        ['colors', 'mask_every_other_prediction'],
+        list(pr_curves_response.keys()))
+
+    # Assert that PR curve data is correct for the colors run.
+    entries = pr_curves_response['colors']
+    self.assertEqual(3, len(entries))
+    self.validatePrCurveEntry(
+        expected_step=0,
+        expected_precision=[0.3333333, 0.3853211, 0.5421686, 0.75, 1.0],
+        expected_recall=[1.0, 0.8400000, 0.3000000, 0.0400000, 0.0],
+        pr_curve_entry=entries[0])
+    self.validatePrCurveEntry(
+        expected_step=1,
+        expected_precision=[0.3333333, 0.3855421, 0.5357142, 0.4000000, 1.0],
+        expected_recall=[1.0, 0.8533333, 0.3000000, 0.0266666, 0.0],
+        pr_curve_entry=entries[1])
+    self.validatePrCurveEntry(
+        expected_step=2,
+        expected_precision=[0.3333333, 0.3934426, 0.5064935, 0.6666666, 1.0],
+        expected_recall=[1.0, 0.8000000, 0.2599999, 0.0266666, 0.0],
+        pr_curve_entry=entries[2])
+
+    # Assert that PR curve data is correct for the mask_every_other_prediction
+    # run.
+    entries = pr_curves_response['mask_every_other_prediction']
+    self.assertEqual(3, len(entries))
+    self.validatePrCurveEntry(
+        expected_step=0,
+        expected_precision=[0.3333333, 0.3786982, 0.5384616, 1.0, 1.0],
+        expected_recall=[1.0, 0.8533334, 0.28, 0.0666667, 0.0],
+        pr_curve_entry=entries[0])
+    self.validatePrCurveEntry(
+        expected_step=1,
+        expected_precision=[0.3333333, 0.3850932, 0.5, 0.25, 1.0],
+        expected_recall=[1.0, 0.8266667, 0.28, 0.0133333, 0.0],
+        pr_curve_entry=entries[1])
+    self.validatePrCurveEntry(
+        expected_step=2,
+        expected_precision=[0.3333333, 0.3986928, 0.4444444, 0.6666667, 1.0],
+        expected_recall=[1.0, 0.8133333, 0.2133333, 0.0266666, 0.0],
+        pr_curve_entry=entries[2])
+
+  def testPrCurvesRaisesValueErrorWhenNoData(self):
+    """Tests that the method for obtaining PR curve data raises a ValueError.
+
+    The handler should raise a ValueError when no PR curve data can be found
+    for a certain run-tag combination.
+    """
+    with self.assertRaises(ValueError):
```

Comment:
You might prefer `with six.assertRaisesRegex(self, ValueError, r'No PR curves could be fetched')` to ensure that the error is the one that you're looking for.